lib/battle/Unit.h

```diff
@@ -26,14 +26,17 @@ class JsonSerializeFormat;
 namespace battle
 {
 
-enum BattlePhases
+namespace BattlePhases
 {
-	SIEGE, // [0] - turrets/catapult,
-	NORMAL, // [1] - normal (unmoved) creatures, other war machines,
-	WAIT_MORALE, // [2] - waited creatures that had morale,
-	WAIT, // [3] - rest of waited creatures
-	MAX_NO_OF_PHASES // [4] - number of phases.
-};
+	enum Type
+	{
+		SIEGE, // turrets/catapult,
+		NORMAL, // normal (unmoved) creatures, other war machines,
+		WAIT_MORALE, // waited creatures that had morale,
+		WAIT, // rest of waited creatures
+		NO_OF_PHASES // number of phases.
```

Comment:
`NO` is bad. `NUMBER_OF_PHASES ` or `PHASE_COUNT` would be better.

Reply:
Updated as per review.